1.2rc1 test failure in astropy.units

This is an unused version of python on this computer, thus the dependencies are far from being up to date. However I think the tests should not fail.
I'm updating numpy to see whether this go away.

```
=========================================================================================================== test session starts ============================================================================================================
platform linux2 -- Python 2.7.6, pytest-2.8.3, py-1.4.30, pluggy-0.3.1

Running tests with Astropy version 1.2rc1.
Running tests in /home/gkovacs/.local/lib/python2.7/site-packages/astropy.

Date: 2016-06-12T13:31:50

Platform: Linux-3.16.0-71-generic-x86_64-with-Ubuntu-14.04-trusty

Executable: /usr/bin/python

Full Python Version: 
2.7.6 (default, Jun 22 2015, 17:58:13) 
[GCC 4.8.2]

encodings: sys: ascii, locale: UTF-8, filesystem: UTF-8, unicode bits: 20
byteorder: little
float info: dig: 15, mant_dig: 15

Numpy: 1.10.1
Scipy: 0.16.0
Matplotlib: not available
h5py: not available
Pandas: not available
```

```
================================================================================================================= FAILURES =================================================================================================================
__________________________________________________________________________________________________ TestQuantityStatsFuncs.test_trace_func __________________________________________________________________________________________________

self = <astropy.units.tests.test_quantity_array_methods.TestQuantityStatsFuncs object at 0x7f81dedaeb50>

    @pytest.mark.xfail(NUMPY_LT_1_10, reason="Numpy 1.10 or later is required")
    def test_trace_func(self):
    
        q = np.array([[1.,2.],[3.,4.]]) * u.m
>       assert np.trace(q) == 5. * u.m
E       assert 5.0 == (5.0 * Unit("m"))
E        +  where 5.0 = <function trace at 0x7f81fd820230>(<Quantity [[ 1., 2.],\n           [ 3., 4.]] m>)
E        +    where <function trace at 0x7f81fd820230> = np.trace
E        +  and   Unit("m") = u.m

/home/gkovacs/.local/lib/python2.7/site-packages/astropy/units/tests/test_quantity_array_methods.py:366: AssertionError
==================================================================================== 1 failed, 10747 passed, 241 skipped, 43 xfailed in 152.41 seconds =====================================================================================
```

--- a/astropy/units/tests/test_quantity_array_methods.py
+++ b/astropy/units/tests/test_quantity_array_methods.py
@@ -5,8 +5,8 @@ import numpy as np
 
 from ... import units as u
 from ...tests.helper import pytest
-from ...utils.compat import (NUMPY_LT_1_8,
-                             NUMPY_LT_1_9_1, NUMPY_LT_1_10)
+from ...utils.compat import (NUMPY_LT_1_8, NUMPY_LT_1_9_1,
+                             NUMPY_LT_1_10, NUMPY_LT_1_10_4)
 
 
 class TestQuantityArrayCopy(object):
@@ -359,7 +359,8 @@ class TestQuantityStatsFuncs(object):
         assert q3.value == np.dot(q1.value, q2.value)
         assert q3.unit == u.m * u.s
 
-    @pytest.mark.xfail(NUMPY_LT_1_10, reason="Numpy 1.10 or later is required")
+    @pytest.mark.xfail(NUMPY_LT_1_10_4,
+                       reason="Numpy 1.10.4 or later is required")
     def test_trace_func(self):
 
         q = np.array([[1.,2.],[3.,4.]]) * u.m

--- a/astropy/utils/compat/numpycompat.py
+++ b/astropy/utils/compat/numpycompat.py
@@ -10,7 +10,7 @@ from ...utils import minversion
 
 
 __all__ = ['NUMPY_LT_1_8', 'NUMPY_LT_1_9', 'NUMPY_LT_1_9_1', 'NUMPY_LT_1_10',
-           'NUMPY_LT_1_11']
+           'NUMPY_LT_1_10_4', 'NUMPY_LT_1_11']
 
 # TODO: It might also be nice to have aliases to these named for specific
 # features/bugs we're checking for (ex:
@@ -19,6 +19,7 @@ NUMPY_LT_1_8 = not minversion('numpy', '1.8.0')
 NUMPY_LT_1_9 = not minversion('numpy', '1.9.0')
 NUMPY_LT_1_9_1 = not minversion('numpy', '1.9.1')
 NUMPY_LT_1_10 = not minversion('numpy', '1.10.0')
+NUMPY_LT_1_10_4 = not minversion('numpy', '1.10.4')
 NUMPY_LT_1_11 = not minversion('numpy', '1.11.0')
 
 

